In [1]:
#Importing libraries
import pandas as pd
import numpy as np
import scipy.stats as stats

# Q1 A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

In [2]:
data=pd.read_csv(r"C:\Users\shrut\Downloads\Cutlets.csv")
data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

H0 (Null Hypothesis):μ1 = μ2 (There is no difference in diameters of cutlets between two units)

H1(Alternative Hypothesis):μ1 ≠ μ2 (There is differnce in diameters of cutlets between two units)

Given 2 indipendent samples and population statndard devaition is not known hence using the 2 sample t test(Individual)

In [4]:
unitA=pd.Series(data.iloc[:,0])
unitA

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [5]:
unitB=pd.Series(data.iloc[:,1])
unitB

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [6]:
p_value=stats.ttest_ind(unitA,unitB)
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

Inference: 
p value(0.4722) > (α = 0.05) Hence we fail to reject the null hypothesis. So there is no significant difference in the diameters of cutlets between the two units.

# Q2 A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

In [8]:
data1=pd.read_csv(r"C:\Users\shrut\Downloads\LabTAT.csv")
data1.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

H0 (Null Hypothesis): There is no difference in the average Turn Around Time (TAT) of reports of the laboratories(No variance).

 H1(Alternative Hypothesis): Atleast one of the average TAT of reports of the laboratories is different(Has variance).

Given 4 independent numeric samples, hence using the ANOVA test.

In [9]:
# Anova ftest statistics: stats.f_oneway(column-1,column-2,column-3,column-4)
p_value=stats.f_oneway(data1.iloc[:,0],data1.iloc[:,1],data1.iloc[:,2],data1.iloc[:,3])
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

Inference:
p value(2.11e-57) < (α = 0.05)

Hence reject the null hypothesis.

So atleast one of the average TAT of reports of the laboratories is different. Thus there is variance or difference in average Turn Around Time (TAT) of reports of the laboratories on their preferred list.

# Q3 Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

In [17]:
data2=pd.read_csv(r"C:\Users\shrut\Downloads\BuyerRatio.csv",index_col=0)
data2.head()

East  West  North  South
Observed Values                          
Males              50   142    131     70
Females           435  1523   1356    750

H0 (Null Hypothesis): male-female buyer ratios are similar across regions(Independent).

H1(Alternative Hypothesis): male-female buyer ratios are not similar across regions(Dependent).

Given the categorical variables, we are checking if there is any association/dependency between the region and the gender, so using Chi-Squared contingency independence test.

In [18]:
table = np.array(data2)
table

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]], dtype=int64)

In [19]:
#Using chi2_contingency test
chi, pval, DOF, expected = stats.chi2_contingency(table)
pval

0.6603094907091882

Inference:

p value(0.66) > (α = 0.05)

Hence, we fail to reject the null hypothesis.

So the male-female buyer ratios are similar across regions(Independent).

# Q4 TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

In [22]:
data3=pd.read_csv(r"C:\Users\shrut\Downloads\Costomer+OrderForm.csv")
data3.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

H0 (Null Hypothesis): The defective % does not varies by centre.

H1(Alternative Hypothesis): The defective % varies by centre.

Given the categorical varibles, So using Chi-Squared contingency independence test.

In [25]:
#Making a contingency table with the given values
data3.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [26]:
data3.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [27]:
data3.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [28]:
data3.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [29]:
table = np.array([[271,267,269,280],[29,33,31,20]])
table

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [30]:
#Using chi2_contingency test
chi, pval, DOF, expected = stats.chi2_contingency(table)
pval

0.2771020991233135

Inference:

p value(0.28) > (α = 0.05)

Hence, we fail to reject the null hypothesis.

Independence of categorical variables Thus, customer order forms defective % does not varies by centre.